index.bs

```diff
@@ -2854,27 +2846,64 @@ visible through the {{WritableStream}}'s public API.
-nothrow>WritableStreamFinishInFlightWrite ( <var>stream</var> )</h4>
+<h4 id="writable-stream-deal-with-rejection" aoid="WritableStreamDealWithRejection"
+nothrow>WritableStreamDealWithRejection ( <var>stream</var>, <var>error</var> )</h4>
 
 <emu-alg>
-  1. Assert: _stream_.[[inFlightWriteRequest]] is not *undefined*.
-  1. <a>Resolve</a> _stream_.[[inFlightWriteRequest]] with *undefined*.
-  1. Set _stream_.[[inFlightWriteRequest]] to *undefined*.
   1. Let _state_ be _stream_.[[state]].
-  1. If _state_ is `"errored"`,
-    1. Perform ! WritableStreamFinishInFlightWriteInErroredState(_stream_).
+  1. If _state_ is `"writable"`,
+    1. Perform ! WritableStreamErroring(_stream_).
     1. Return.
-  1. Assert: _state_ is `"writable"`.
-  1. Perform ! WritableStreamHandleAbortRequestIfPending(_stream_).
+  1. Assert: _state_ is `"erroring"`.
+  1. Perform ! WritableStreamFinishError(_stream_).
 </emu-alg>
 
-<h4 id="writable-stream-finish-in-flight-write-in-errored-state" aoid="WritableStreamFinishInFlightWriteInErroredState"
-nothrow>WritableStreamFinishInFlightWriteInErroredState ( <var>stream</var> )</h4>
+<h4 id="writable-stream-erroring" aoid="WritableStreamErroring" nothrow>WritableStreamErroring ( <var>stream</var>,
+<var>reason</var> )</h4>
 
 <emu-alg>
-  1. Perform ! WritableStreamRejectAbortRequestIfPending(_stream_).
-  1. Perform ! WritableStreamRejectPromisesInReactionToError(_stream_).
+  1. Assert: _stream_.[[storedError]] is *undefined*.
+  1. Assert: _stream_.[[state]] is `"writable"`.
+  1. Let _controller_ be _stream_.[[writableStreamController]].
+  1. Assert: _controller_ is not *undefined*.
+  1. Set _stream_.[[state]] to `"erroring"`.
+  1. Set _stream_.[[storedError]] to _reason_.
+  1. Let _writer_ be _stream_.[[writer]].
+  1. If _writer_ is not *undefined*, perform !
+     WritableStreamDefaultWriterEnsureReadyPromiseRejected(_writer_, _reason_).
+  1. If ! WritableStreamHasOperationMarkedInFlight(_stream_) is *false* and _controller_.[[started]] is *true*, perform
+     ! WritableStreamFinishError(_stream_).
+</emu-alg>
+
+<h4 id="writable-stream-finish-error" aoid="WritableStreamFinishError" nothrow>WritableStreamFinishError
+( <var>stream</var> )</h4>
+
+<emu-alg>
+  1. Assert: _stream_.[[state]] is `"erroring"`.
+  1. Assert: ! WritableStreamHasOperationMarkedInFlight(_stream_) is *false*.
+  1. Set _stream_.[[state]] to `"errored"`.
+  1. Perform ! _stream_.[[writableStreamController]].[[ErrorSteps]]().
+  1. Let _storedError_ by _stream_.[[storedError]];
+  1. Repeat for each _writeRequest_ that is an element of _stream_.[[writeRequests]],
+    1. <a>Reject</a> _writeRequest_ with _storedError_.
+  1. Set _stream_.[[writeRequest]] to an empty List.
+  1. if _stream_.[[pendingAbortRequest]] is *undefined*,
+    1. Perform !  WritableStreamRejectCloseAndClosedPromiseIfNeeded(_stream_).
+    1. Return.
+  1. Let _abortRequest_ by _stream_.[[pendingAbortRequest]].
```

Comment:
I found one of these myself which means I had it wrong in three places. 'y' and 'e' aren't even close to each other on the keyboard. Very odd.